var/spack/repos/builtin/packages/py-astrovascpy/package.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyAstrovascpy(PythonPackage):
+    """
+    Vasculature blood flow computation and impact of astrocytic
+    endfeet on vessels
+    """
+
+    homepage = "https://github.com/BlueBrain/AstroVascPy"
+    git = "https://github.com/BlueBrain/AstroVascPy.git"
+    pypi = "AstroVascPy/AstroVascPy-0.1.5.tar.gz"
+
+    license("Apache-2.0")
+
+    maintainers("tristan0x")
+
+    version("0.1.5", sha256="9f444775f464de740590f8acd880caae87ca1ffdf7d524490d6b8ce21bb4d5af")
+
+    variant("vtk", default=False, description="add VTK support (mainly for visualization)")
+
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-setuptools-scm", type="build")
+
+    depends_on("py-click", type=("build", "run"))
+    depends_on("py-cached-property", type=("build", "run"))
+    depends_on("py-coverage", type=("build", "run"))
+    depends_on("py-cython", type=("build", "run"))
+    depends_on("py-h5py+mpi", type=("build", "run"))
+    depends_on("py-libsonata", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
+    depends_on("py-mpi4py", type=("build", "run"))
+    depends_on("py-morphio", type=("build", "run"))
+    depends_on("py-networkx", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-pandas", type=("build", "run"))
+    depends_on("py-petsc4py", type=("build", "run"))
+    depends_on("py-psutil", type=("build", "run"))
+    depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-seaborn", type=("build", "run"))
+    depends_on("py-tables", type=("build", "run"))
+    depends_on("py-tqdm", type=("build", "run"))
+    depends_on("py-trimesh", type=("build", "run"))
+    depends_on("py-vascpy", type=("build", "run"))
+    depends_on("vtk+python", type=("build", "run"), when="+vtk")
+
+    depends_on("py-pytest", type="test")
+    depends_on("py-pytest-mpi", type="test")
```

Comment:
@tristan0x I find it highly unlikely that such large package these work with all versions of those packages. For example, https://github.com/BlueBrain/AstroVascPy/blob/main/pyproject.toml already states minimum versions of `setuptools` and `setuptools_scm`.
https://github.com/BlueBrain/AstroVascPy/blob/main/setup.py has `python_requires=">=3.10",`

If you can, please contact the upstream project and ask them for the minimum versions that they test with for their dependencies and ask them to add them to their requirements. I did not see that they declare this full requirements and supported Python versions in their pyproject.toml, which they should really do.

I guess, if they are not responsive, you may add the minimum versions that you built or tested with, add them as minimum versions to the depends on and add a comment that these are what have been tested by you.